include/thread_pool/mpmc_bounded_queue.hpp

```diff
@@ -76,18 +86,23 @@ class MPMCBoundedQueue
     MPMCBoundedQueue& operator=(MPMCBoundedQueue&& rhs) noexcept;
 
     /**
-     * @brief push Push data to queue.
-     * @param data Data to be pushed.
-     * @return true on success.
-     */
+    * @brief MPMCBoundedQueue destructor.
+    */
+    virtual ~MPMCBoundedQueue() = default;
```

Comment:
Is this class ever derived from?  Are there any other virtual methods?  It may be worth marking the class as final and removing the virtual dtor.

Reply:
Fixed!